File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

n = int(sys.stdin.readline()) # 컴퓨터 수
f = int(sys.stdin.readline()) # 네트워크 상에서 연결되어 있는 수

l = [[] for _ in range(n + 1)] # 그래프 표시
visited = [0] * (n + 1) # 방문 컴퓨터 표시
answer = 0 # 정답

# 그래프 표시
for i in range(f):
    x, y = map(int, sys.stdin.readline().split())
    l[x].append(y)
    l[y].append(x)

# 깊이 우선 탐색으로 문제 해결
def dfs(k):
    global answer
    if not visited[k]:
        visited[k] = True
        for i in l[k]:
            if not visited[i]:
                answer += 1
                dfs(i)


dfs(1)

print(answer)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

public class Main{
    // attribute
    public static int[][] arr;
    public static boolean[] visited;
    public static int answer;

    public static void main(String[] args) throws Exception{
        // 입력값
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int number = Integer.parseInt(br.readLine());
        arr = new int[number+1][number+1];
        visited = new boolean[number + 1];
        int n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n ; i ++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            // save - 양방향
            arr[x][y] = 1;
            arr[y][x] = 1;
        }

        // 출력값
        dfs(1, number);
        System.out.println(answer);
    }

    public static void dfs(int number, int size){
        visited[number] = true;
        for (int i = 1 ; i < size + 1 ; i++){
            if(visited[i] || arr[number][i] == 0) continue;
            answer++;
            dfs(i, size);
        }
    }
}